lib/source-node.js

```diff
@@ -151,7 +151,7 @@ SourceNode.fromStringWithSourceMap =
       if (mapping === null || mapping.source === undefined) {
         node.add(code);
       } else {
-        var source = aRelativePath
+        var source = aRelativePath && mapping.source
```

Comment:
I suspect that in the case where `aRelativePath` is non-null and `mapping.source` is null, that this should evaluate to `aRelativePath`.

So maybe this ought to be fixed in `util.join` instead; that function already tries to handle the empty string, changing it to handle `null` and/or `undefined` as well seems reasonable.

What do you think?

Reply:
That makes sense. I will change that and push up the new change soon.

Reply:
This pr has been open for a long time, what is the current status? is it still relevant?

Reply:
I think the bug still exists, but this patch still needs an update and a test case.